metpy/cbook.py

```diff
@@ -39,4 +37,103 @@ def get_test_data(fname, as_file_obj=True):
     return path
 
 
-__all__ = ('get_test_data', 'is_string_like', 'iterable')
+class CacheData(object):
+    def __init__(self):
```

Comment:
How about this takes a `path` argument, defaulting to `None`. Then:
```python
self.cache_home = self.default_location if path is None else path
if not os.path.exists(self.cache_home):
    os.makedirs(self.cache_home)
```

Taking it explicitly makes this a more re-usable class, and I think it will allow us to test better.